# 异常

**捕获异常语法**

```py
try:
    # 执行代码
    ...
except OSError as err:
    ...
except ValueError:
    ...
except (RuntimeError, TypeError, NameError): # 支持处理多个异常
    ...
except:
    # 发生异常时执行的代码
    ...
else:
    # 未发生异常时执行的代码
    ...
finally:
    # 不论是否发生异常都执行的代码
    ...
```

**抛出异常语法**

```py
raise [Exception [, args [, traceback]]]
```

**示例**

```py
try:
    raise NameError('HiThere')  # 模拟一个异常。
except NameError:
    print('An exception flew by!')
    raise
```